I'm just trying to get this persons string segmenter to work as mine …

shannonfromomaha opened the pull request:
…did and it won't even rackup

the error code is far to long to post in here, but if someone could look at it and tell me why it won't even open I would appreciate it.

shannonfromomaha committed:
I'm just trying to get this persons string segmenter to work as mine did and it won't even rackup

the error code is far to long to post in here, but if someone could look at it and tell me why it won't even open I would appreciate it.

sumeetjain commented on app/controllers/main_controller.rb:
```diff
@@ -1,5 +1,7 @@
+require_relative '..lib/string_segmenter.rb'
```
I know Alex is helping you, but would changing this to `'../lib/string_segmenter.rb'` make a difference?

shannonfromomaha committed:
fixed the problems to get myself back to ZERO

shannonfromomaha committed:
I'm getting the following error "The valid data types are: binary, boolean, date, datetime, decimal, float, integer, primary_key, references, string, text, time, timestamp" for my define_schema file.
However, I did make models of "Inputstring" and "Exportstring" (like Sumeet did earlier with "Director" and "Movie")...am I missing a step in the video that would allow me to use these?

halfghaninne commented:
`define_column` is actually a method and it takes three arguments: table name, column name, and data type. Looks like you are passing in two arguments.

shannonfromomaha committed:
SO CLOSE! I'm having trouble with my main_controller--instead of sending anything to /searches it assumes you are trying to segment "searches"! is this a fatal flaw with how my program runs or is there something i can do to help this?
ps i have to leave at 430 today but if i don't get in the queue by then i can respond to a pr comment and still definitely get done by 6

sumeetjain commented:
This is a situation where different HTTP methods (get vs. post, for example) would be useful. Since we're not there yet, I suggest using different paths to make the controller actions less confusing for Sinatra.

shannonfromomaha committed:
despite having used the example code, it seems my data base isn't saving to the "searches" table...and i'm not sure why

StevenThiesfeld commented:
Your controller methods seem a bit jumbled so I would recommend the following.  Your welcome page's form should submit (the action = '/blank') to what looks like /:str which should be named without the colon, as that is only needed if you're passing parameters through the URL.  The form will make that unnecessary.  Keep in mind that whatever you name the text field in your form will be the key for that value in the params hash you access in the controller.

shannonfromomaha commented:
i think if you're seeing the /:str it's an old commit--sorry. this should be the newest one (the one with the example code in it)

StevenThiesfeld commented:
It looks like you have multiple models defined.  You should only have one model (class inheriting from ActiveRecord) per table, you don't need one for each column within the table.

shannonfromomaha committed:
done--although i will have a question tomorrow about how you can empty a stub database
